src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -295,7 +297,7 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService, TextAs
             if (result != null && !isAtLeastInstructorForExercise && result.getAssessor() != null && !result.getAssessor().getLogin().equals(user.getLogin())
                     && result.getCompletionDate() == null) {
                 // If we already have a result, we need to check if it is locked.
-                throw new BadRequestAlertException("This submission is being assessed by another tutor", ENTITY_NAME, "alreadyAssessed");
+                return forbidden(ENTITY_NAME, "alreadyAssessed", "This submission is being assessed by another tutor");
```

Comment:
Just a remark here, I'm not sure if what I'm thinking is correct. The user is technically authorized to access the resource (as a tutor), so maybe the 403 Forbidden is not the correct status code. The case here is that the lock has been granted to another tutor, so perhaps 423 Locked would be the better choice.